### PR TITLE
Fixes sm slivers not properly deleting

### PR DESCRIPTION
**Pull request by harmonyn**
* Fixes the issue with the sliver not qdelling properly enabling mass murder. Fixes https://github.com/tgstation/tgstation/issues/28710.

### CODE REVIEW

File: code/game/objects/items/theft_tools.dm [hunk 1]
```diff
@@ -14,6 +14,7 @@
 	var/pulse = 0
 	var/cooldown = 0
 	var/pulseicon = "plutonium_core_pulse"
+	var/list/blacklisted_locs = list(/obj/machinery/nuclearbomb,  /obj/item/nuke_core_container)
```
Comment (lzimann) [inline code/game/objects/items/theft_tools.dm]: Rename this to `var/static/list/blacklisted_locs_typecache = typecacheof(list(/obj/machinery/nuclearbomb,  /obj/item/nuke_core_container))`

File: code/game/objects/items/theft_tools.dm [hunk 2]
```diff
@@ -30,6 +31,9 @@
 		return ..()
 
 /obj/item/nuke_core/process()
+	var/turf/T = loc
+	if(T in blacklisted_locs)	// no radiation when in proper containment
```
Comment (lzimann) [inline code/game/objects/items/theft_tools.dm]: Remove the T, replace this with `if(is_type_in_typecache(loc, blacklisted_locs_typecache)`

**Review by lzimann — CHANGES_REQUESTED** [review]

File: code/game/objects/items/theft_tools.dm [hunk 3]
```diff
@@ -215,6 +219,8 @@
 		return
 	if(ismovableatom(O))
 		Consume(O)
+		QDEL_NULL(sliver)
+		to_chat(usr, "<span class='notice'>\The [sliver] is dusted along with \the [O]!</span>")
```
Comment (ChangelingRain) [inline code/game/objects/items/theft_tools.dm]: move this above the QDEL_NULL

**Comment by ShizCalev** [comment]
also fixes #28688 i'd assume?

**Comment by bgobandit** [comment]
yeah didn't know that was an issue good work players

File: code/game/objects/items/theft_tools.dm [hunk 4]
```diff
@@ -31,8 +31,7 @@
 		return ..()
 
 /obj/item/nuke_core/process()
-	var/turf/T = loc
-	if(T in blacklisted_locs)	// no radiation when in proper containment
+	if(is_type_in_typecache(loc, blacklisted_locs_typecache)
```
Comment (ShizCalev) [inline code/game/objects/items/theft_tools.dm]: missing a ) at the end.

File: code/game/objects/items/theft_tools.dm [hunk 5]
```diff
@@ -30,6 +31,8 @@
 		return ..()
 
 /obj/item/nuke_core/process()
+	if(is_type_in_typecache(loc, blacklisted_locs_typecache))
+		return
 	if(cooldown < world.time - 60)
```
Comment (ShizCalev) [inline code/game/objects/items/theft_tools.dm]: just curious, should this be else if instead now?

**Comment by Really-Good-Soda-Flavor** [comment]
Remove the nuke cores fix, you've done it poorly. Nuclear bombs simply had to be moved to Initialize().

**Review by lzimann — APPROVED** [review]

**Pull request merged**